src/main/java/com/icthh/xm/ms/balance/service/BalanceService.java

```diff
@@ -314,6 +302,31 @@ public TransferDto transfer(TransferBalanceRequest transferRequest) {
             .build();
     }
 
+    private BalanceChangeEvent createBalanceChangeEvent(String operationUuid, BigDecimal amount, Metadata metadata,
+                                                   Instant now, Balance balance, boolean isSubtractAmount,
+                                                   OperationType transferTo) {
+        BigDecimal amountBeforeTransferTo = balanceRepository.findBalanceAmount(balance).orElse(ZERO);
+        BigDecimal amountAfterTransferTo = getAmountAfter(isSubtractAmount, amountBeforeTransferTo, amount);
+        Optional<BalanceChangeEvent> lastBalanceChangeEvent = balanceChangeEventRepository.findLastBalanceChangeEvent(balance.getId());
```

Comment:
seems you do not need separate variable `lastBalanceChangeEvent`, so do not break optional processing chain (find().map().orElse()).

Reply:
but, it's this way more readable

Reply:
Do you really think that having the intermediate variable `lastBalanceChangeEvent` is more readable than that : ?
```
Instant prevOperationDate = balanceChangeEventRepository
      .findLastBalanceChangeEvent(balance.getId())
      .map(BalanceChangeEvent::getOperationDate)
      .orElse(Instant.EPOCH);
```

Reply:
Fixed